File: src/fkt/java/util/Objects.java
package fkt.java.util;
import java.util.ArrayList;
/**
Utility methods for arrays. 
 */
public final class Objects{
  private final static boolean debug=false;
	public static String toString(Object[]items,String spacer){
	  if(items==null)return "null";
	  else if(items.length==0)return "";
	  ArrayList list=new ArrayList();
		boolean trim=false&&!spacer.equals("\n");
		int at=0;
	  for(Object item:items)list.add(
				(item==null?"null":trim?item.toString().trim():item)
				+(++at==items.length?"":spacer)
			);
	  return list.toString();
	}
	public static String toString(Object[]array){
		return toString(array,",");
	}
	public static String toLines(Object[]array){
	  if(array==null)return "null";
	  ArrayList list=new ArrayList();
	  for(int i=0;i<array.length;i++)
	    list.add((array[i]==null?"null"
	    		:true?Debug.info(array[i]):array[i].toString())
	    	+(i<array.length-1?"\n":""));
	  String lines=list.toString();
		return false?lines:lines.replaceAll("\n"," ");
	}
}

File: src/fkt/facets/util/Objects.kt
package fkt.facets.util
import java.util.ArrayList
/**
 * Utility methods for arrays.
 */
object Objects {
  private fun toString(items: Array<Any?>?, spacer: String): String {
    if (items == null)
      return "null"
    else if (items.isEmpty()) return ""
    val list = ArrayList<Any>()
    val trim = false && spacer != "\n"
    var at = 0
    for (item in items)
      list.add(
        (if (item == null) "null" else if (trim) item.toString().trim { it <= ' ' } else item).toString() + if (++at == items.size) "" else spacer
      )
    return list.toString()
  }

  fun toString(array: Array<Any?>): String {
    return toString(array, ",")
  }

  fun toLines(array: Array<*>?): String {
    if (array == null) return "null"
    val list = ArrayList<Any>()
    for (i in array.indices)
      list.add((if (array[i] == null)
        "null"
      else if (true) Debug.info(array[i]) else array[i].toString()) + if (i < array.size - 1) "\n" else "")
    val lines = list.toString()
    return if (false) lines else lines.replace("\n".toRegex(), " ")
  }
}
